fix(components): [menu] trigger submenu focus

esttom opened the pull request:
Fixed an issue where the focus remains when a submenu is selected on mouseover

closed #11640

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

esttom committed:
fix(components): [menu] trigger submenu focus

Fixed an issue where the focus remains when a submenu is selected on mouseover

closed #11640

tolking commented on packages/components/menu/src/sub-menu.ts:
```diff
@@ -256,6 +257,7 @@ export default defineComponent({
 
       timeout?.()
       ;({ stop: timeout } = useTimeoutFn(() => {
+        submenuRef.value?.focus()
```
It is better to trigger before the animation.

```suggestion
      if (
        rootMenu.isMenuPopup &&
        event.type === 'mouseenter' &&
        rootMenu.props.menuTrigger === 'hover'
      ) {
        submenuRef.value?.focus()
        }
```

esttom replied:
Thanks for the review.
I have corrected the code.
Please confirm it.

esttom committed:
fix(components): [menu] trigger submenu focus

Fixed an issue where the focus remains when a submenu is selected on mouseover

closed #11640
